lib/ddtrace/contrib/ethon/multi_patch.rb

```diff
@@ -0,0 +1,39 @@
+require 'ddtrace/ext/net'
+require 'ddtrace/ext/distributed'
+require 'ddtrace/propagation/http_propagator'
+require 'ddtrace/contrib/ethon/ext'
+
+module Datadog
+  module Contrib
+    module Ethon
+      # Ethon MultiPatch
+      module MultiPatch
+        def self.included(base)
+          # No need to prepend here since add method is included into Multi class
+          base.send(:include, InstanceMethods)
+        end
+
+        # InstanceMethods - implementing instrumentation
+        module InstanceMethods
+          def add(easy)
+            handles = super(easy)
+            return handles if handles.nil? || !tracer_enabled?
+
+            easy.datadog_before_request
```

Comment:
Not very familiar with the construction of the framework, but is `Multi` composed of several `Easy` instances? And when it executes, will it call `Easy#perform`? If so, would that call `datadog_before_request` making this redundant?

Reply:
Per my response above, is there an equivalent `#perform` method for `Multi`? Might want to consider adding a parent span around the lifetime of this multi operation which bundles up the `Easy` requests. Just a thought.

Reply:
I added the parent span support. The logic is the following: it is created when the first `easy` is added to `multi` and it is finalized when `multi#perform` has finished. `Multi` can be reused - every time `easy` is added it creates a parent span if it doesn't exist yet.